macros/supporting/datatypes.sql

```diff
@@ -0,0 +1,19 @@
+{# string  -------------------------------------------------     #}
+
+{%- macro type_string() -%}
+  {{ adapter_macro('dbtvault.type_string') }}
+{%- endmacro -%}
+
+{% macro default__type_string() %}
+    {%- if execute -%}
+        {{ exceptions.raise_compiler_error("The 'dbtvault.type_string' macro does not support your database engine. Supported Databases: Snowflake, Bigquery") }}
+    {%- endif -%}
+{% endmacro %}
+
+{%- macro bigquery__type_string() -%}
+    STRING
+{%- endmacro -%}
+
+{% macro snowflake__type_string() %}
+    VARCHAR
```

Comment:
@DVAlexHiggs so it turns out that this STRING and VARCHAR distinction was the primary reason I had a helper function to do a cast.

Since this is gonna keep popping up with different db vendors, I thought we should probably start a type macro set 

for example, as an extension  if you start using INTs I'll have to use INT64 or something etc etc